### PR TITLE
Delete account feature

### PR DESCRIPTION
**Pull request by Peter0892**
Added delete account feature with alert feedback to allow the users to understand what went wrong in the event that it is not possible for them to delete their account.

### CODE REVIEW

File: src/providers/auth.service.ts [hunk 1]
```diff
@@ -121,4 +122,27 @@ export class AuthService {
...
+          data => {
+            observer.next(data);
+            observer.complete();
+          },
+          err => {
+            observer.next(err);
+            observer.complete();
+          },
+          () => {
+            observer.complete();
```
Comment (JTalarico) [inline src/providers/auth.service.ts]: Why are there 3 parameters? There should just be response and error.

File: src/providers/auth.service.ts [hunk 2]
```diff
@@ -121,4 +122,27 @@ export class AuthService {
...
+  /**
+   * Delete user
+   *
+   * @returns {Observable}
+   */
+  public deleteUser() {
+    return new Observable((observer) => {
+      this.authHttp.delete(environment.server_url + "/api/user")
+        .subscribe(
+          data => {
```
Comment (JTalarico) [inline src/providers/auth.service.ts]: This should be res not data, to stay consistent with the rest of the code.

File: src/providers/auth.service.ts [hunk 3]
```diff
@@ -17,7 +17,8 @@ export class AuthService {
   constructor(private authHttp: AuthHttp,
               public zone: NgZone,
               public storage: Storage,
-              public http: Http) {
+              public http: Http,
+  ) {
 
     // Check if there is a profile saved in local storage
```
Comment (JTalarico) [inline src/providers/auth.service.ts]: No inline comments.

File: src/pages/edit-account/edit-account.ts [hunk 4]
```diff
@@ -67,4 +74,78 @@ export class EditAccountPage {
...
+
+  /**
+   * Close the modal page.
+   */
+  private dismiss() {
+    this.viewCtrl.dismiss().catch( (f) => f);
+  }
+
+  /**
+   * Handle user deletion.
```
Comment (JTalarico) [inline src/pages/edit-account/edit-account.ts:121]: More description. What happens on failure? What does this do.

File: src/pages/edit-account/edit-account.ts [hunk 5]
```diff
@@ -67,4 +74,78 @@ export class EditAccountPage {
...
+            this.auth.logout();
+
+            this.navCtrl.popToRoot();
+            this.navCtrl.push(LoginPage);
+
+            this.dismiss();
+            this.presentToast();
+          } else {
+            let alertButtons = [
+              {
```
Comment (JTalarico) [inline src/pages/edit-account/edit-account.ts:140]: Why does this need a variable? Can't you just have
Alert(this.alertCtrl, "Delete User Failed", response.errors.message, [{role: "cancel", text: "Dismiss"}]);

**Review by JTalarico — CHANGES_REQUESTED** [review]

File: src/pages/edit-account/edit-account.ts (hunk 5 above)
Comment (Peter0892) [reply src/pages/edit-account/edit-account.ts:140]: Using a variable helps clean the code and works for better code readability in my opinion. In this case it does not affect performance either.

File: src/providers/auth.service.ts (hunk 1 above)
Comment (Peter0892) [reply src/providers/auth.service.ts]: The proper RxJS way of doing the subscription based on their documentation is:   
x => console.log('onNext: %s', x), 
e => console.log('onError: %s', e), 
() => console.log('onCompleted'));

where x is onNext, e is onError and () is onCompleted function handlers

Comment (JTalarico) [reply src/providers/auth.service.ts]: Fair enough, however we have not been using on complete so just use on next and on error.

**Review by JTalarico — APPROVED** [review]

**Pull request merged**